+nosnoc/+discrete_time_problem/Stewart.m

```diff
@@ -376,29 +410,49 @@ function generate_direct_transcription_constraints(obj)
                     obj.f = obj.f + dcs.f_q_fun(x_ijk, z_ijk, lambda_ijk, theta_ijk, mu_ijk, u_i, v_global, p);
                 end
 
-                % Clock <Constraints
+                % Clock Constraints
                 % TODO(@anton) HERE BE DRAGONS. This is by far the worst part of current nosnoc as it requires the discrete problem
-                %              to understand something about the
-                %              time-freezing reformulation which is ugly. (It is also just a quite complicated thing)
-                if obj.opts.use_fesd && opts.equidistant_control_grid
-                    if opts.time_optimal_problem
-                        if opts.use_speed_of_time_variables
-                            obj.g.equidistant_control_grid(ii) = {[sum_h - opts.h;s_sot*sum_h - obj.w.T_final()/opts.N_stages]};
-                        else
-                            obj.g.equidistant_control_grid(ii) = {sum_h - obj.w.T_final()/opts.N_stages};
-                        end
+                %              to understand something about the time-freezing reformulation which is ugly.
```

Comment:
we should judge less the public code and have neutral tone comments either explaining stuff, or adding remarks (for TODOs we better have issues, as they are easier to find). 
also, the time options are just quite complicated.

Reply:
#117 replaces these comments.